kubectl-plugin/pkg/cmd/delete/delete.go

```diff
@@ -0,0 +1,152 @@
+)
+
+type DeleteOptions struct {
+	configFlags  *genericclioptions.ConfigFlags
+	ioStreams    *genericiooptions.IOStreams
+	ResourceType util.ResourceType
+	ResourceName string
+	Namespace    string
+}
+
+var deleteExample = templates.Examples(`
+		# Delete RayCluster
+		kubectl ray delete raycluster sample-raycluster
+
+		# Delete RayJob
+		kubectl ray delete rayjob sample-rayjob
+
+		# Delete RayService
+		kubectl ray delete rayservice sample-rayservice
+	`)
+
+func NewDeleteOptions(streams genericiooptions.IOStreams) *DeleteOptions {
+	configFlags := genericclioptions.NewConfigFlags(true)
+	return &DeleteOptions{
+		ioStreams:   &streams,
+		configFlags: configFlags,
+	}
+}
+
+func NewDeleteCommand(streams genericclioptions.IOStreams) *cobra.Command {
+	options := NewDeleteOptions(streams)
+	factory := cmdutil.NewFactory(options.configFlags)
+
+	cmd := &cobra.Command{
+		Use:          "delete (TYPE NAME)",
+		Short:        "Delete Ray resoruce.",
+		Example:      deleteExample,
+		Long:         `Deletes Ray custom resources such as RayCluster, RayService, or RayJob`,
+		SilenceUsage: true,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if err := options.Complete(cmd, args); err != nil {
+				return err
+			}
+			if err := options.Validate(); err != nil {
+				return err
+			}
+			return options.Run(cmd.Context(), factory)
+		},
+	}
+
+	options.configFlags.AddFlags(cmd.Flags())
+	return cmd
+}
+
+func (options *DeleteOptions) Complete(cmd *cobra.Command, args []string) error {
+	if len(args) != 2 {
+		return cmdutil.UsageErrorf(cmd, "%s", cmd.Use)
+	}
+
+	if *options.configFlags.Namespace == "" {
+		options.Namespace = "default"
+	} else {
+		options.Namespace = *options.configFlags.Namespace
+	}
+
+	switch strings.ToLower(args[0]) {
+	case string(util.RayCluster):
+		options.ResourceType = util.RayCluster
+	case string(util.RayJob):
+		options.ResourceType = util.RayJob
+	case string(util.RayService):
+		options.ResourceType = util.RayService
+	default:
+		return cmdutil.UsageErrorf(cmd, "unsupported resource type: %s", args[0])
+	}
+
+	options.ResourceName = args[1]
+
+	return nil
+}
+
+func (options *DeleteOptions) Validate() error {
+	// Overrides and binds the kube config then retrieves the merged result
+	config, err := options.configFlags.ToRawKubeConfigLoader().RawConfig()
+	if err != nil {
+		return fmt.Errorf("Error retrieving raw config: %w", err)
+	}
+	if len(config.CurrentContext) == 0 {
+		return fmt.Errorf("no context is currently set, use %q to select a new one", "kubectl config use-context <context>")
+	}
+	return nil
+}
+
+func (options *DeleteOptions) Run(ctx context.Context, factory cmdutil.Factory) error {
+	k8sClient, err := client.NewClient(factory)
+	if err != nil {
+		return fmt.Errorf("failed to create client: %w", err)
+	}
+
+	// Ask user for confirmation
+	reader := bufio.NewReader(os.Stdin)
+	fmt.Printf("Are you sure you want to delete %s %s? (Y/y/yes) ", options.ResourceType, options.ResourceName)
```

Comment:
I think the message should be `(y/n)` or `(y/yes/n/no)` instead of only showing yes to users.

Reply:
Done.